edge/pkg/metamanager/client/configmap.go

```diff
@@ -51,24 +52,33 @@ func (c *configMaps) Delete(string) error {
 
 func (c *configMaps) Get(name string) (*api.ConfigMap, error) {
 	resource := fmt.Sprintf("%s/%s/%s", c.namespace, model.ResourceTypeConfigmap, name)
-	configMapMsg := message.BuildMsg(modules.MetaGroup, "", modules.EdgedModuleName, resource, model.QueryOperation, nil)
-	msg, err := c.send.SendSync(configMapMsg)
-	if err != nil {
-		return nil, fmt.Errorf("get configmap from metaManager failed, err: %v", err)
-	}
-	errContent, ok := msg.GetContent().(error)
-	if ok {
-		return nil, errContent
-	}
-	content, err := msg.GetContentData()
-	if err != nil {
-		return nil, fmt.Errorf("parse message to configmap failed, err: %v", err)
+	metas, err := dao.QueryMeta("key", resource)
+	if err != nil || len(*metas) == 0 {
```

Comment:
According to the logic here, if the `configmap` information can be queried in the database, it will not be obtained from the cloud, right? But if there is a problem with the information in the database and an error occurs in the subsequent `handleConfigMapFromMetaDB` parsing,  according to the above logic, we also cannot execute the part of obtaining data from the cloud because database have configmap information(Even the data in the database is wrong)?

Reply:
> According to the logic here, if the `configmap` information can be queried in the database, it will not be obtained from the cloud, right? But if there is a problem with the information in the database and an error occurs in the subsequent `handleConfigMapFromMetaDB` parsing, according to the above logic, we also cannot execute the part of obtaining data from the cloud because database have configmap information(Even the data in the database is wrong)?

When the configmap changes in the cloud, it will be updated to the edge sqlite. Prioritizing obtaining it from the edge sqliet can solve the problem of poor cloud edge network.

Reply:
@luomengY just to be sure about this fallback behavior.

> When the configmap changes in the cloud, it will be updated to the edge sqlite.

is there any racy situation that this might get `not-up-to-date` configmap from edge sqlite then? in other word, edge and cloud configmap is always synced, so that edge cannot get `not-up-to-date` configmap from edge sqlite at all?

i was thinking what we would want to do as fallback behavior, maybe we can try cloud and if response is none (e.g network connectivity or anything), then we can fall back to call edge sqlite? this guarantees that edge can get the latest configmap and usually this case just works unless network connectivity problems?